services/resources-getter.js

```diff
@@ -63,66 +65,88 @@ function ResourcesGetter(model, opts, params) {
   }
 
   function getWhere() {
-    var where = {};
-    where[OPERATORS.AND] = [];
+    return new P(function (resolve, reject) {
+      var where = {};
+      where[OPERATORS.AND] = [];
 
-    if (params.search) {
-      where[OPERATORS.AND].push(new SearchBuilder(model, opts, params,
-          fieldNamesRequested).perform());
-    }
+      if (params.search) {
+        where[OPERATORS.AND].push(new SearchBuilder(model, opts, params,
+            fieldNamesRequested).perform());
+      }
 
-    if (params.filter) {
-      where[OPERATORS.AND].push(handleFilterParams());
-    }
+      if (params.filter) {
+        where[OPERATORS.AND].push(handleFilterParams());
+      }
 
-    if (segmentWhere) {
-      where[OPERATORS.AND].push(segmentWhere);
-    }
+      if (segmentWhere) {
+        where[OPERATORS.AND].push(segmentWhere);
+      }
 
-    return where;
+      if (params.query) {
+        var rawQuery = params.query.trim();
+        new LiveQueryChecker().perform(rawQuery);
+
+        opts.connections[0]
```

Comment:
What if the connection for the collection is the first one?

Reply:
I've added the same comment than in the query-stat-getter